unit-tests/log/test-multi-thread-logs.cpp

```diff
@@ -0,0 +1,231 @@
+    }
+    avg_time_ms /= 50;
+    avg_value /= 50;
+    return std::make_pair(avg_time_ms, avg_value);
+}
+
+//----------------------- HELPER METHODS END ---------------------------
+
+TEST_CASE("async logger", "[log][async_log]")
+{
+    size_t n_callbacks = 0;
+    auto callback = [&](rs2_log_severity severity, rs2::log_message const& msg)
+    {
+        ++n_callbacks;
+        std::cout << msg.raw();
+    };
+    
+    auto func = [](int required_value) {
+        int iterations = 0;
+        auto start = std::chrono::steady_clock::now();
+        
+        while (iterations < number_of_iterations)
+        {
+            std::stringstream ss;
+            ss << "atomic integer = " << ++atomic_integer;
+            auto start_ms = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch());
+            rs2::log(RS2_LOG_SEVERITY_DEBUG, ss.str().c_str());
+            std::chrono::milliseconds ms = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch());
+            std::chrono::milliseconds delta_ms = ms - start_ms;
+            if (delta_ms > max_time) max_time = delta_ms;
+            if (delta_ms < min_time) min_time = delta_ms;
+            avg_time += delta_ms;
+            start_ms = ms;
+            ++iterations;
+        }
+    };
+    const char* log_file_path = ".//multi-thread-logs.log";
+    try {
+        remove(log_file_path);
+    }
+    catch (...) {}
+    rs2::log_to_file(RS2_LOG_SEVERITY_DEBUG, log_file_path);
+
+    std::vector<std::thread> threads;
+
+    for (int i = 0; i < number_of_threads; ++i)
+    {
+        threads.push_back(std::thread(func, i + 1));
+    }
+    
+    for (auto&&t : threads)
+    {
+        if (t.joinable()) t.join();
+    }
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(5000));
+    std::stringstream ss;
+    ss << "max time = " << max_time.count() << " ms" << std::endl;
+    ss << "min time = " << min_time.count() << " ms" << std::endl;
+    ss << "avg time = " << (float) (avg_time.count()) / (number_of_iterations * number_of_threads) << " ms" << std::endl;
+    CAPTURE(ss.str());
+    CHECK_NOFAIL(max_time.count() < checked_log_write_time);
+    REQUIRE(max_time.count() < required_log_write_time);
+    
+    // checking logs correctness - that all the integer's values have been logged
+    {
+        // preparing array of number of iterations * number of threads
+        std::array<char, number_of_iterations * number_of_threads> check_table;
+        // setting all cells to 0
+        memset(&check_table[0], 0, check_table.size());
+        // marking cells at the position of each log integer value to 1
+        std::ifstream check_file(log_file_path, std::ios::in);
+        if (check_file.good())
+        {
+            std::string line;
+            while (std::getline(check_file, line)) {
+                auto value_in_log = stoi(line.substr(line.find_last_of(" ")));
```

Comment:
1. What if ` ` wasn't found?
2. Don't you want the index + 1?

Reply:
1. This helper method is adapted to the test only. There is a space in the logs I use in the test. 
2. It worked also without the "+1" because the space was deleted in the stoi. I have added it.